### PR TITLE
core(driver): add run warning on page load timeout

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Adds a toplevel warning when we reach the page load timeout. Apparently some of us thought we were already doing this anyway :)

**Related Issues/PRs**
#10527

### CODE REVIEW

File: lighthouse-core/gather/gather-runner.js [hunk 1]
```diff
@@ -24,6 +24,11 @@ const UIStrings = {
   warningRedirected: 'The page may not be loading as expected because your test URL ' +
   `({requested}) was redirected to {final}. ` +
   'Try testing the second URL directly.',
+  /**
+   * @description Warning that Lighthouse timed out while waiting for the page to load.
+   */
+  warningTimeout: 'The page loaded too slowly to finish within the maximum time limit. ' +
```
Comment (connorjclark) [inline lighthouse-core/gather/gather-runner.js:30]:
```suggestion
  warningTimeout: 'The page loaded too slowly to finish within the time limit. ' +
```

File: readme.md [hunk 2]
```diff
@@ -273,9 +273,9 @@ yarn build-all
 ```
 
 #### installing protobuf
-If changing audit output, you'll need to have v3.7.1 of the protocol-buffer/protobuf compiler installed. (v3.7.1 is known to be compatible, and 3.11.x is known to be **not** compatible.). 
+If changing audit output, you'll need to have v3.7.1 of the protocol-buffer/protobuf compiler installed. (v3.7.1 is known to be compatible, and 3.11.x is known to be **not** compatible.).
 
-Homebrew should be able to install it correctly: `brew install protobuf@3.7.1`
+Homebrew should be able to install it correctly: `brew install protobuf@3.7`
```
Comment (connorjclark) [inline readme.md:278]: were these changes on purpose?

Comment (patrickhulce) [reply readme.md:278]: yes drive by fix as I needed to run this on my new machine and `brew install protobuf@3.7.1` doesn't work, it only supports @3.7

File: lighthouse-core/gather/driver.js [hunk 3]
```diff
@@ -1120,14 +1129,18 @@ class Driver {
       /* eslint-enable max-len */
 
       if (!waitForFcp) maxFCPMs = undefined;
-      await this._waitForFullyLoaded(pauseAfterFcpMs, pauseAfterLoadMs, networkQuietThresholdMs,
-        cpuQuietThresholdMs, maxWaitMs, maxFCPMs);
+      const loadPromise = await this._waitForFullyLoaded(pauseAfterFcpMs, pauseAfterLoadMs,
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/gather/driver.js:1132]:
```suggestion
      const loadResult = await this._waitForFullyLoaded(pauseAfterFcpMs, pauseAfterLoadMs,
```

File: lighthouse-core/gather/driver.js [hunk 4]
```diff
@@ -1120,14 +1129,18 @@ class Driver {
...
+        networkQuietThresholdMs, cpuQuietThresholdMs, maxWaitMs, maxFCPMs);
+      timedOut = loadPromise.timedOut;
```
Comment (patrickhulce) [inline lighthouse-core/gather/driver.js:1134]:
```suggestion
      timedOut = loadResult.timedOut;
```

File: readme.md (hunk 2 above)
Comment (brendankenny) [reply readme.md:278]: > yes drive by fix as I needed to run this on my new machine and `brew install protobuf@3.7.1` doesn't work, it only supports @3.7

Looks like `@3.7` is [the official way to refer to the older version now](https://formulae.brew.sh/formula/protobuf)

(our tests not working as desired on the lastest protobuf release it going to come around to bite us sooner or later...:)

**Comment by connorjclark** [comment]
Still LGTM, i'll let you land it tho.
